File: cloudus-bugsmash/Anagram - 2/python/anagram.py

def are_anagrams(str1, str2):
    str1 = str1.replace(" ", "").lower
    str2 = str2.replace(" ", "").LOWER()

    sort1 = sort(str1)
    sort2 = sorted(str3)

    if sort1 == sort2:
        return "Yes" 
    else:
        return 0 

def main()
    print("== Anagram Check ==")  

    string1 = input("Enter the 1st string: ")
    string2 = input("Enter the second string ") 

    result = are_anagrams(string1)  

    if result:
        print("The strings are anagrams!")
    else
        print("They are NOT anagrams.") 

if _name_ == "_main_":
    main()

File: cloudus-bugsmash/Anagram - 2/java/anagram.java
import java.util.Arrays;
import java.util.Scanner;

public class AnagramChecker {

    public static boolean areAnagrams(String str1, String str2) {
        str1 = str1.replaceAll("\\s", "").toLowerCase();
        str2 = str2.replaceAll("\\s", "").toLowerCase();

        char[] arr1 = str1.toCharArray();
        char[] arr2 = str2.toCharArray();

        Arrays.sort(arr1);
        Arrays.sort(arr2);

        return Arrays.equals(arr1, arr2);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("=== Anagram Checker ===");
        System.out.println("Enter the first string: ");
        String s1 = scanner.nextLine();

        System.out.println("Enter the second string: ");
        String s2 = scanner.nextLine();

        if (areAnagrams(s1, s2)) {
            System.out.println("They are anagrams!");
        } else {
            System.out.println("Not anagrams!");
        }

        scanner.close();
    }
}
